app/importers/unpaywall_publication_importer.rb

```diff
@@ -0,0 +1,91 @@
+# frozen_string_literal: true
+
+class UnpaywallPublicationImporter
+  def import_all
+    unless Rails.env.test?
+      pbar = ProgressBar.create(title: 'Importing publication data from Unpaywall',
+                                total: all_pubs.count)
+    end
+
+    all_pubs.find_each do |p|
+      query_unpaywall_for(p)
+      pbar.increment unless Rails.env.test?
+    end
+    pbar.finish unless Rails.env.test?
+  end
+
+  def import_new
+    unless Rails.env.test?
+      pbar = ProgressBar.create(title: 'Importing publication data from Unpaywall',
+                                total: new_pubs.count)
+    end
+
+    new_pubs.find_each do |p|
+      query_unpaywall_for(p)
+      pbar.increment unless Rails.env.test?
+    end
+    pbar.finish unless Rails.env.test?
+  end
+
+  private
+
+    def all_pubs
+      Publication.where.not(doi: nil).where.not(doi: '')
+    end
+
+    def new_pubs
+      all_pubs.where(unpaywall_last_checked_at: nil)
+    end
+
+    def get_pub(url)
+      attempts = 0
+      HTTParty.get(url).to_s
+    rescue Net::ReadTimeout, Net::OpenTimeout
+      if attempts <= 10
+        attempts += 1
+        retry
+      else
+        raise
+      end
+    end
+
+    def query_unpaywall_for(publication)
+      unpaywall_json = nil
+      find_url = URI::DEFAULT_PARSER.escape("https://api.unpaywall.org/v2/#{publication.doi_url_path}?email=openaccess@psu.edu")
+      unpaywall_json = JSON.parse(get_pub(find_url))
+
+      best_oa_location = unpaywall_json.dig('best_oa_location', 'url')
+      existing_oa_location = publication.open_access_locations.find_by(source: 'Unpaywall')
+      oa_status = unpaywall_json['oa_status']
+
+      if best_oa_location
+        if existing_oa_location
+          existing_oa_location.update!(url: best_oa_location)
+        else
+          publication.open_access_locations.create!(source: 'Unpaywall', url: best_oa_location)
+        end
+      else
+        existing_oa_location.try(:destroy)
+      end
+
+      publication.open_access_status = oa_status
+      publication.unpaywall_last_checked_at = Time.current
```

Comment:
`Time.zone.now`

Reply:
I'm fine with making this switch, but just curious what the benefit is. It looks like we are using both pretty equally throughout the app.

Reply:
@nicktiberi You should _always_ use `Time.zone` in Rails projects, eg `Time.zone.now` and `Time.zone.parse` and `Time.zone.local`. Using `Time.new` will only cause you pain and suffering when all your stuff breaks +/- 5 hours from midnight. 

The reason is that by default Ruby does not take timezones into account, and therefore `Time.new` is blissfully unaware of what a timezone is. Our production servers are running in UTC time, and everything stored in the database is stored in UTC time (as it should be). However, our application code is configured to run in Eastern Time (as it should be). Therefore we need to use `Time.zone` when we work with dates and times, to attach the current timezone to the Datetime object, so it will be properly translated from eastern time, into utc, and back again.

This doesn't cause any issues when you run your tests on your development machine, because your system time zone (Eastern) matches the application's configured time, but in ci and on production this can lead to bugs and broken tests.

tldr; never never never use `Time.new` and `Time.now`, always use `Time.zone._____` in Rails apps

Sincerely, 
Lord of The Time Zones, who once literally got into a fight with NOAA/NASA (and won) about the correct handling of timezones.